Sources/Nimble/Matchers/Equal.swift

```diff
@@ -12,7 +12,9 @@ public func equal<T: Equatable>(_ expectedValue: T?) -> Predicate<T> {
             return PredicateResult(status: .fail, message: msg)
         case (let expected?, let actual?):
             let matches = expected == actual
-            return PredicateResult(bool: matches, message: msg)
+            let compactFailMsg = ExpectationMessage.fail(diff(expected, actual).joined(separator: "\n"))
```

Comment:
do we need this in other `equal` methods?

Reply:
yes, we do. there are a couple of dozens in Antares. In others probably too.
There is another overload for sets, which require handling as well.

Reply:
In the set overload I deleted quite a few lines. Take a look in Jira how it affects the output.